packages/ui/src/components/sign-up/indentifiers.tsx

```diff
@@ -0,0 +1,27 @@
+import { useClerk } from '@clerk/clerk-react';
+
+import { parsePhoneString } from '~/common/phone-number-field/utils';
+import type { RequireExactlyOne } from '~/types/utils';
+
+type Identifiers = {
+  emailAddress: boolean;
+  phoneNumber: boolean;
+};
+
+type Identifier = RequireExactlyOne<Identifiers>;
+
+export function SignUpIdentifier({ emailAddress, phoneNumber }: Identifier) {
+  const { client } = useClerk();
+  const signUpEmailAddress = client.signUp.emailAddress;
+  const { formattedNumberWithCode: signUpPhoneNumber } = parsePhoneString(client.signUp.phoneNumber || '');
```

Comment:
❔ Minor thing feel free to ignore, should we consider moving the parsing here inside the if statement below to avoid calling `parsePhoneString` when there is no phone number?

Reply:
Good idea https://github.com/clerk/javascript/pull/3694/commits/ca02d78723f105f4e49b694981863048f71a9ec3